Sofa/Component/Engine/Select/src/sofa/component/engine/select/BaseROI.inl

```diff
@@ -0,0 +1,779 @@
+                const bool is_in_roi = (strict) ? isQuadInStrictROI(q) : isQuadInROI(q);
+                if (is_in_roi)
+                {
+                    quadIndices.push_back(i);
+                    quadInROI.push_back(q);
+                }
+                else
+                {
+                    quadOutIndices.push_back(i);
+                    quadsOutROI.push_back(q);
+                }
+            }
+        }
+
+        //Tetrahedra
+        if (d_computeTetrahedra.getValue())
+        {
+            for(unsigned int i=0 ; i<tetrahedra.size() ; i++)
+            {
+                const auto& t = tetrahedra[i];
+                const bool is_in_roi = (strict) ? isTetrahedronInStrictROI(t) : isTetrahedronInROI(t);
+                if (is_in_roi)
+                {
+                    tetrahedronIndices.push_back(i);
+                    tetrahedraInROI.push_back(t);
+                }
+                else
+                {
+                    tetrahedronOutIndices.push_back(i);
+                    tetrahedraOutROI.push_back(t);
+                }
+            }
+        }
+
+        //Hexahedra
+        if (d_computeHexahedra.getValue())
+        {
+            for(unsigned int i=0 ; i<hexahedra.size() ; i++)
+            {
+                const auto& t = hexahedra[i];
+                const bool is_in_roi = (strict) ? isHexahedronInStrictROI(t) : isHexahedronInROI(t);
+                if (is_in_roi)
+                {
+                    hexahedronIndices.push_back(i);
+                    hexahedraInROI.push_back(t);
+                }
+                else
+                {
+                    hexahedronOutIndices.push_back(i);
+                    hexahedraOutROI.push_back(t);
+                }
+            }
+        }
+
+        d_nbIndices.setValue(sofa::Size(indices.size()));
+    }
+}
+
+
+template <class DataTypes>
+void BaseROI<DataTypes>::draw(const core::visual::VisualParams* vparams)
+{
+    if(d_componentState.getValue() == ComponentState::Invalid)
+        return ;
+
+    if (!vparams->displayFlags().getShowBehaviorModels())
+        return;
+
+    const VecCoord& x0 = d_X0.getValue();
+    constexpr auto color = sofa::type::RGBAColor(1.0f, 0.4f, 0.4f, 1.0f);
+
+    vparams->drawTool()->saveLastState();
+
+    if (d_drawROI.getValue())
+    {
+        roiDraw(vparams);
+    }
+
+    const float sizeFactor = std::max(static_cast<float>(d_drawSize.getValue()), 1.0f);
+
+    vparams->drawTool()->setLightingEnabled(false);
+
+    constexpr auto convertToVertex = [](const auto& coord) -> type::Vec3d{
+
+        if constexpr(DataTypes::spatial_dimensions == 1)
+        {
+            return { coord[0], 0.0, 0.0};
+        }
+        if constexpr (DataTypes::spatial_dimensions == 2)
+        {
+            return { coord[0], coord[1], 0.0 };
+        }
+        if constexpr (DataTypes::spatial_dimensions >= 3)
+        {
+            return { coord[0], coord[1], coord[2] };
+        }
```

Comment:
```suggestion
        return coord;
```
Not tested, but I believe that Vec can handle any type of size conversion.

Reply:
at least on msvc, the vertices.push_back() did not let setting with the CPos directly 🫨